ci-operator/step-registry/idp/external-oidc/keycloak/aks/server/idp-external-oidc-keycloak-aks-server-commands.sh

```diff
@@ -0,0 +1,395 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+export KUBECONFIG="${SHARED_DIR}/kubeconfig"
+
+# Insall helm binary
+wget https://get.helm.sh/helm-v3.17.2-linux-amd64.tar.gz -O /tmp/helm.tar.gz
+tar -xvf /tmp/helm.tar.gz -C /tmp/
+chmod +x /tmp/linux-amd64/helm
+
+KEYCLOAK_ADMIN_TEST_USER="admin-$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 6 | head -n 1 || true)"
+KEYCLOAK_ADMIN_TEST_PASSWORD="$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 12 | head -n 1 || true)"
+KEYCLOAK_PREFIX="keycloak-$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 9 | head -n 1 || true)"
+KEYCLOAK_HOST="$KEYCLOAK_PREFIX.$HYPERSHIFT_BASE_DOMAIN"
+CLUSTER_CONSOLE="$(oc whoami --show-console)"
+CONSOLE_CLIENT_SECRET_VALUE=$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 32 | head -n 1 || true)
+
+# Generate random test users for testing
+TEST_USERS=""
+for i in $(seq 1 50);
+do
+    username="keycloak-testuser-${i}"
+    password=$(< /dev/urandom tr -dc 'a-z0-9' | fold -w 12 | head -n 1 || true)
+    TEST_USERS+="${username}:${password},"
+done
+# The users will be mounted into keycloak pods and the real test users will be created there
+TEST_USERS=${TEST_USERS%?}
+
+# Install NGINX Ingress Controller
+/tmp/linux-amd64/helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+/tmp/linux-amd64/helm repo update
+/tmp/linux-amd64/helm install ingress-nginx ingress-nginx/ingress-nginx \
+  --namespace ingress-nginx \
+  --create-namespace \
+  --set controller.service.annotations."service\.beta\.kubernetes\.io/azure-load-balancer-health-probe-request-path"=/healthz \
+  --wait
+
+# Install cert-manager
+oc create namespace cert-manager || true
+oc apply -f https://github.com/jetstack/cert-manager/releases/download/v1.13.0/cert-manager.yaml
+oc wait --for=condition=ready pod -l app.kubernetes.io/instance=cert-manager -n cert-manager --timeout=300s
+# Create Let's Encrypt ClusterIssuer
+cat <<EOF | oc apply -f -
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-prod
+spec:
+  acme:
+    server: https://acme-v02.api.letsencrypt.org/directory
+    email: openshift-qe@redhat.com
+    privateKeySecretRef:
+      name: letsencrypt-prod
+    solvers:
+    - http01:
+        ingress:
+          class: nginx
+EOF
+oc wait --for=condition=ready clusterissuer letsencrypt-prod --timeout=120s
+
+# Set Keycloak repo
+/tmp/linux-amd64/helm repo add bitnami https://charts.bitnami.com/bitnami
+/tmp/linux-amd64/helm repo update
+
+# Create Keycloak settings file
+cat > /tmp/keycloak-values.yaml << EOF
+nameOverride: ""
+fullnameOverride: ""
+namespace: keycloak
+
+auth:
+  adminUser: ${KEYCLOAK_ADMIN_TEST_USER}
+  adminPassword: "${KEYCLOAK_ADMIN_TEST_PASSWORD}"
+image:
+  registry: docker.io
```

Comment:
docker.io has limit rate on anonymous pulls. Pulls might occasionally fail, right? If so, is there a way to resolve it?